src/tests/test_viewsGenerationSystem.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (C) 2021 NV Access Limited
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+"""
+Runs the dataView generation system on test data.
+Confirms the end result data is as expected.
+Creates a number of specific scenarios for the test data.
+"""
+
+from dataclasses import dataclass
+from typing import Iterable
+import unittest
+
+
+@dataclass
+class TestSet:
+	inputDir: str
+	outputDir: str
+
+
+class _GenerateSystemTestData:
+	"""
+	A class which generates system test data for a given testing set given.
+	"""
+	def __init__(self, testSet: TestSet) -> None:
+		pass
+
+	def generate_testData(self):
+		"""
+		Generates system test data.
+		"""
+		self.test_addon_to_be_downgraded()
+		self.test_addon_versions()
+		self.test_stable_from_beta()
+		self.test_addon_to_be_fully_removed()
+		self.test_NVDAVersions()
+
+	def test_addon_to_be_fully_removed(self):
+		"""Creates addon data for an addon to be fully removed in subsequent datasets"""
+		pass
+
+	def test_addon_to_be_downgraded(self):
+		"""Creates addon data for an addon to be downgraded in subsequent datasets"""
+		pass
+
+	def test_addon_versions(self):
+		"""Creates addon data to test the newest available addon version is used"""
+		pass
+
+	def test_stable_from_beta(self):
+		"""Generates a beta addon for the first set, and the equivalent stable addon for the second
+		"""
+		pass
+
+	def test_NVDAVersions(self):
+		"""Creates addon data to test that the newest addon is used across various NVDA versions"""
+		pass
```

Comment:
I'd prefer to see unit tests for the system before system tests. Unit tests are less fragile. If the most of the system can be covered by unit tests, then only a few (perhaps one) system tests is need to ensure everything is glued together correctly.

At this stage, the exact details of "what you will test" aren't as important as "how you will test"

Reply:
I've added some of the unit tests. I thought the main idea here was to just show that the unit testing system works.

Reply:
This has been done. See the commits 6935464, ba2f951